ntoskrnl/ps/security.c

```diff
@@ -689,25 +691,38 @@ PsImpersonateClient(IN PETHREAD Thread,
             return STATUS_UNSUCCESSFUL;
         }
 
-        /* Make sure we can impersonate */
-        if (!SeTokenCanImpersonate(ProcessToken,
-                                   Token,
-                                   ImpersonationLevel))
+        /*
+         * The server may want to obtain identification details of a client
+         * instead of impersonating so just give the server a pass.
+         */
+        if (Level > SecurityIdentification)
```

Comment:
Shouldn't this inequality check better be done by `SeTokenCanImpersonate` ? (It takes `Level` as parameter, and original code already tried to do validation checks on `Level`.)